src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -49,7 +53,18 @@ public User add(User user) {
             query.setParameter("email", email);
             return Optional.ofNullable((User) query.uniqueResult());
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get user", e);
+            throw new DataProcessingException("Can't get user by email", e);
+        }
+    }
+
+    @Override
+    public Optional<User> getById(Long userId) {
+        try (Session session = sessionFactory.openSession()) {
+            Query query = session.createQuery("FROM User WHERE userId = :userId");
+            query.setParameter("userId", userId);
+            return Optional.ofNullable((User) query.uniqueResult());
```

Comment:
```suggestion
            return Optional.ofNullable(query.uniqueResult());
```
try to avoid casting

Reply:
done